src/content/plugins/chunks-webpack-plugin.md

```diff
@@ -0,0 +1,322 @@
+---
+title: ChunksWebpackPlugin
+contributors:
+  - yoriiis
+related:
+  - title: SplitChunksPlugin
+    url: /plugins/split-chunks-plugin
+  - title: The real power of Webpack 4 SplitChunks Plugin
+    url: https://medium.com/@Yoriiis/the-real-power-of-webpack-4-splitchunks-plugin-fad097c45ba0
+---
+
+The ChunksWebpackPlugin creates HTML files with entry points and chunks relations to serve your webpack bundles. It is suitable with multiple page application which contains multiple entry points.
+
+Since Webpack 4, SplitChunksPlugin offer the possibility to optimizes all chunks. It can be particularly powerful, because it means that chunks can be shared even between async and non-async chunks. See the webpack documentation for [`splitChunks.chunks`](https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkschunks).
+
+This option automatically generate new chunks associated with an entry point. For example, entry points `a.js` and `b.js` share common codes with the file `vendors~a~b.js`.
+
+With multiple entry points, it can be difficult to identify relation between auto-generated chunks and entry points.
+
+The plugin parse the `entrypoints` [Map](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Map) object from the webpack compilation to get all valid entry points and associated files. Then, it generates HTML files which include all assets filtered by entry point and the `chunks-manifest.json` file.
+
+## Zero config
+
+The plugin works without configuration. For advanced usage, see the section [using configuration](#using-a-configuration).
+
+## Installation
+
+The plugin is available as a package with the name of `chunks-webpack-plugin` on [npm](https://www.npmjs.com/package/chunks-webpack-plugin) and [Github](https://github.com/yoriiis/chunks-webpack-plugin).
+
+```bash
+npm install chunks-webpack-plugin --save-dev
+```
+
+```bash
+yarn add chunks-webpack-plugin --dev
+```
+
+## Environment
+
+ChunksWebpackPlugin was built for Node.js `>=8.11.2` and webpack `>=4.x`.
+
+## Example
+
+For example implementation, please see the [Github documentation](https://github.com/yoriiis/chunks-webpack-plugin/blob/master/README.md#example).
+
+## Basic usage
+
+The plugin will generate two HTML files for each entry points. Each filename contains the entry point name, the `{{entry}}` placeholder is automatically replaced.
+
+- `{{entry}}-styles.html`: contains all HTML `<link>` tags
+- `{{entry}}-scripts.html`: contains all HTML `<script>` tags
+
+First, let's add the plugin to the webpack configuration.
+
+```js
+var ChunksWebpackPlugin = require('chunks-webpack-plugin');
+var path = require('path');
+
+module.exports = {
+  entry: 'main.js',
+  output: {
+    filename: 'main.js',
+    path: path.resolve(__dirname, './dist')
+  },
+  plugins: [new ChunksWebpackPlugin()]
+};
+```
+
+HTML files are built in the output path directory with all the other assets.
+
+Then, include the HTML files in the wanted pages.
```

Comment:
What do you mean? Are you describing what plugin does or what user has to do?

Reply:
The first line (L69) describe what plugin does.
The second line (L71) describe what user has to do to use this generated HTML files.

HTML files are generated by the plugin but need to be includes by the user in the wanted pages. True, I have not described this part because I didn't want to add a language dependencies like the `include` in PHP or Twig.

But we can clarify this part if you want.

Reply:
ok i think i understand, maybe go for this or along these lines:
```suggestion
Now you can include the outputted HTML files into your HTML page templates. You can do it with e.g. Twig or PHP.
```